CMSgov/easi-app
# EASI-2204 Feedback v2: Remove old survey references

# EASI-2204

Please check the ticket for related details and comments

This pr only covers component display toggling over the `helpFooter` flag.

## Changes and Description

- Uses `helpFooter` launchdarkly flag
- Put old survey references behind the flag `!flags.helpFooter`
  Generally suggested by `surveymonkey.com` links
  - AnythingWrongSurvey
  - ImproveEasySurvey
  - ReportProblemLinkArea
    - REPORT_PROBLEM_ACCESSIBILITY_TEAM_SURVEY
    - REPORT_PROBLEM_BASIC_USER_SURVEY

## Testing

- Set flag `helpFooter: true`
- Visit pages of affected components and see that the links are gone

### Testing notes

The `flags` object is `{}` in jest tests.

Affected snapshots have survey output removed. Affected unit tests are skipped and will be removed after the feedback changeover is all done.

### Examples

Report a problem
| Before | After |
| --- | --- |
| ![Screen Shot 2022-08-17 at 12 02 47 PM](https://user-images.githubusercontent.com/97050498/185190616-f2bf1900-11c0-4b95-8950-89caddca684c.png) | ![Screen Shot 2022-08-17 at 12 02 58 PM](https://user-images.githubusercontent.com/97050498/185190696-3d33c022-59a1-40bf-9586-716b6868a380.png) |

Help us improve
| Before | After |
| --- | --- |
| ![Screen Shot 2022-08-17 at 12 03 51 PM](https://user-images.githubusercontent.com/97050498/185190824-4bf85da5-cddb-4c97-bd8e-ca9984aee305.png) | ![Screen Shot 2022-08-17 at 12 04 03 PM](https://user-images.githubusercontent.com/97050498/185190860-a7dd8dfa-853d-4106-9365-ae1ddafa4388.png) |
